src/main/java/com/hazelcast/cloud/model/UserSerializer.java

```diff
@@ -0,0 +1,32 @@
+package com.hazelcast.cloud.model;
+
+import com.hazelcast.cloud.model.User;
+import com.hazelcast.nio.serialization.compact.CompactReader;
+import com.hazelcast.nio.serialization.compact.CompactSerializer;
+import com.hazelcast.nio.serialization.compact.CompactWriter;
+
+public class UserSerializer implements CompactSerializer<User> {
```

Comment:
Do you really need `UserSerializer`? Compact serialization is turned on by default and it worked well for the `City` without serialization.

Reply:
It doesn't serialize to `City` object. It's just generic record. If there is no serialization, it will be generic record which is the case. And, we want to promote SQL and Compact.

Reply:
But you specify `valueFormat' = 'compact'` when creating mapping. Isn't it enough?
```
        String mappingSql = ""
            + "CREATE OR REPLACE MAPPING users("
            + "CREATE OR REPLACE MAPPING cities("
            + "     __key INT,"
            + "     name VARCHAR,"
            + "     country VARCHAR"
            + "     country VARCHAR,"
            + "     city VARCHAR,"
            + "     population INT"
            + ") TYPE IMap"
            + " OPTIONS ("
            + "     'keyFormat' = 'int',"
            + "     'valueFormat' = 'compact',"
            + "     'valueCompactTypeName' = 'user'"
            + "     'valueCompactTypeName' = 'city'"
            + " )";
```

Reply:
No, if we don't provide the serializer explicitly, it will use the zero-config serializer, and the type name will be "com.hazelcast.cloud.model.City" (which is confusing and long) and it will be hard to write equivalent serializers in different client languages (they have to use the same long type name as well). I believe having an explicit serializer is the better approach here

Reply:
Well I'm not sure about other languages, but you are writing the code for Java. Why should you care about other languages?
Also I'm not sure but isn't type name specified in mapping with `'valueCompactTypeName' = 'city'`

Reply:
Yes, but you might very well have a Python client connected to the same cluster, and they should be interoperable. 

Regarding the other question, it is not working like this. When you don't have the explicit serializer, the type name in the serialized data will be `com.hazelcast.cloud.model.City`, and when you say the type name is `city` while creating the mapping, these two will be essentially different schemas, not compatible with each other. Besides, writing serializer is our preferred way of using Compact

Reply:
From my point of view zero-config serializer should be the preferred way as it doesn't require writing of the boring code and have duplication with the mapping creation. But you can go your way if you think that usage of compact serializer is more commonly used approach.